framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4631,6 +4613,442 @@ private void makeGroundTargetType(
     addDefaultAnnotations(functionalType);
   }
 
+  /**
+   * Return true if {@code type} should be captured.
+   *
+   * <ul>
+   *   <li>{@code type} and {@code typeMirror} must both be declared types.
```

Comment:
These conditions are introduced without preamble or segue.  I first interpreted them as method preconditions, but after reading the implementation it seems more likely that they are laying out the method's algorithm.  (Neither can be correct, because for this line, violation of one of the conditions leads to `return false` but violation of the other leads to `throw new BugInCF`.)  Could you please clarify?